pandas/tests/dtypes/test_common.py

```diff
@@ -16,44 +16,46 @@ class TestPandasDtype(object):
 
     # Passing invalid dtype, both as a string or object, must raise TypeError
     # Per issue GH15520
-    def test_invalid_dtype_error(self):
-        msg = 'not understood'
-        invalid_list = [pd.Timestamp, 'pd.Timestamp', list]
-        for dtype in invalid_list:
-            with tm.assert_raises_regex(TypeError, msg):
-                com.pandas_dtype(dtype)
-
-        valid_list = [object, 'float64', np.object_, np.dtype('object'), 'O',
-                      np.float64, float, np.dtype('float64')]
-        for dtype in valid_list:
+    @pytest.mark.parametrize('dtype', [pd.Timestamp, 'pd.Timestamp', list])
+    def test_invalid_dtype_error(self, dtype):
+        with tm.assert_raises_regex(TypeError, 'not understood'):
             com.pandas_dtype(dtype)
 
-    def test_numpy_dtype(self):
-        for dtype in ['M8[ns]', 'm8[ns]', 'object', 'float64', 'int64']:
-            assert com.pandas_dtype(dtype) == np.dtype(dtype)
+    @pytest.mark.parametrize('dtype', [
+        object, 'float64', np.object_, np.dtype('object'), 'O',
+        np.float64, float, np.dtype('float64')])
+    def test_pandas_dtype_valid(self, dtype):
+        assert com.pandas_dtype(dtype) == dtype
```

Comment:
Not sure this test is actually necessary?  Split this off from the `test_invalid_dtype_error`, as it appears to be testing _valid_ instead of invalid dtypes.  Previously this didn't have any type of `assert` statement; it just called `pandas_dtype` and the test passed if no error was raised.

Reply:
yeah this is ok,  this is just testing that pandas_dtype doesn't raise on numpy dtypes